src/NLog/Config/StackTraceUsage.cs

```diff
@@ -33,36 +33,56 @@
 
 namespace NLog.Config
 {
+    using System;
+
     /// <summary>
     /// Value indicating how stack trace should be captured when processing the log event.
     /// </summary>
+    [Flags]
     public enum StackTraceUsage
     {
         /// <summary>
         /// Stack trace should not be captured.
         /// </summary>
-        None = 0, 
+        None = 0,
+
+        /// <summary>
+        /// Stack trace should be captured
+        /// </summary>
+        WithStackTrace = 1,
```

Comment:
maybe: WithFullStackTrace?

Reply:
What is the difference between "full stacktrace" and "stacktrace" ?

Reply:
it's to make the difference between stacktrace and callsite more clear. But OK, this isn't a good proposal